edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesSpecFactory.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes
+{
+    using System.Collections.Generic;
+    using Newtonsoft.Json;
+
+    public class KubernetesSpecFactory<CombinedDockerConfig> : IKubernetesSpecFactory<CombinedDockerConfig>
```

Comment:
What is a `spec` that you introduced here? How it is different from `EdgeDeployment`?

Reply:
EdgeDeployment has always had a "Spec". It's the custom part of the custom resource.